docs/validate.rst

```diff
@@ -48,8 +50,10 @@ classes should adhere to.
 
     .. attribute:: DEFAULT_TYPES
 
-        The default mapping of JSON types to Python types used when validating
-        :validator:`type` properties in JSON schemas.
+        Deprecated. Under normal usage, this will be an empty dictionary.
```

Comment:
We can't make this an empty dictionary, it has to preserve the existing behavior at least for `Draft3Validator` and `Draft4Validator`, and also for anyone who is using the `default_types` argument to `create`. If tests are missing for that we probably need to add them, it's existing behavior that someone could be relying on, so we can deprecate it but can't change it until we remove all those arguments.

We also need to have accessing the attribute warn with one of the warnings you've added (thanks!)

We can either use a property for that, or look at one of the libraries that are focused around "deprecation" itself if it starts to get annoying.

Reply:
Good catch. This is partly a wording issue but I'm also going to revisit that in the code. Currently `default_types` is set to the old default if no `TypeChecker` is provided. However, you made me realise that this always needs to be true, because of direct access to the property.

Reply:
> We also need to have accessing the attribute warn with one of the warnings you've added (thanks!)

This isn't quite a trivial as first sounds - `DEFAULT_TYPES` is a class property and access to it is very likely to be done so on the class rather than a concrete instance.

So we _could_ do one of the fudges for class properties, but is it worth it? Any access to `DEFAULT_TYPES` is likely (and arguably only a problem) to be followed by some derived use (i.e. `create(default_types=...)` or `validator(types=...)`), both of which would trigger a deprecation warning.